src/authenticators/ldap_ynhuser.py

```diff
@@ -201,6 +202,13 @@ def _authenticate_credentials(self, credentials=None):
         except ValueError:
             raise YunohostError("invalid_credentials")
 
+        username = ldap.filter.escape_filter_chars(username)
+        # Search username, if user give a mail instead
+        if "@" in username:
+            user = _get_ldap_interface().search("ou=users", f"mail={username}", ["uid"])
```

Comment:
@alexAubin  do you think there is a security risk to expose the _get_ldap_interface during authentication ?

Reply:
Hmmm idk, why do you think it could be a security risk ? :thinking: I don't quite see what kind of injection could happen there

Reply:
Don't know how this API should be isolated or not from the other part of the code that was written with hypothesis that the API user is an admin...

I don't remember well how is made _get_ldap_interface and if we could imagine injection through username  (or other strange thing like attack on memory).

Reply:
Nah it's fine, the portal API runs as non-root and our LDAP ACL are decent

`_get_ldap_interface()` will only open an anonymous connection in the context of the portal, cf : https://github.com/YunoHost/yunohost/blob/dev/src/utils/ldap.py#L90

Worst case scenario, the attacker could find some way to inject stuff via the provided username (but `escape_filter_chars` seems to prevent that i suppose ?) and maybe enumerate / leak the whole list of existing username maybe but that's about it

Reply:
Ok so i mark this as solved